modules/drivers/sqlite/src/metabase/driver/sqlite.clj

```diff
@@ -360,6 +361,97 @@
   [_]
   (hsql/call :datetime (hx/literal :now)))
 
+(defmethod sql.qp/->honeysql [:sqlite :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x                (sql.qp/->honeysql driver x)
+        y                (sql.qp/->honeysql driver y)
+        extract          (fn [unit x] (sql.qp/date :sqlite unit x))
+        disallowed-types (keep
+                          (fn [v]
+                            (some->> v
+                                     hx/type-info
+                                     hx/type-info->db-type
+                                     name
+                                     (re-find #"(?i)^time")))
+                          [x y])]
+    (when (seq disallowed-types)
+      (throw (ex-info (tru "Only datetime, timestamp, or date types allowed. Found {0}"
```

Comment:
Same suggestion as the other PR: you should mention that those types are only allowed **for `datetimeDiff()**. Otherwise people could get pretty confused by this error. If the error stuff was abstracted out into a helper function then you'd only have to change the error message in one place🙃